nav2_msgs/action/ComputeControl.action

```diff
@@ -1,5 +1,6 @@
 #goal definition
 nav_msgs/Path path
+string controller_name
```

Comment:
`string motion_type`?

Reply:
This is the string in the map of controller plugins to use. If we call the key the controller name and the value is the controller, ei `FollowPath: DWBLocalPlanner::RegisteredPluginType` then the field in the action message should also be the name. 

Motion type doesn't make clear to me this is the same name as the map in the controller server. The eq. message in the planner uses `planner_name` so there's some automatic parallels  someone can draw if you know one, you know both. I'd be open to other names but I'd like them to be consistent between the 2 servers in style

Reply:
My comment was directed towards the fact that we were opening this message to mean any type of control/actuation, but if this is `FollowPath.action`, then yes, `motion_type` is not needed. 

However, I actually liked naming this action to mean *moving the robot's base*. This would then allow to specify the type of motion as `FollowPath`, `SuperSafeButSlowFollowPath`, `RoughlyFollowPath`, etc. We don't know/care at this level about the types of controllers (e.g. `DWB`, `MPC`, `PID`), only about the high-level intention. The matching of the motion type string to the plugin name would be done down on the `ControllerServer` and not hardcoded but on the params file.

I don't see how exposing the names of the controllers at this level provides more *contextual control*. I could argue that my proposal provides actually more.

Reply:
Controller name is FollowPath, SuperSafeButSlowFollowPath, ... its the `name` of the `controller` stored in the `controllers_` map and the parameter in the configuration file of `controller_names`. 

Again, that's what's happening here as we've discussed. You are defining a **map** of plugins not a vector. The level of abstraction is clear. If you make the name of your controller FollowPath and map to MPC and FollowPathSlowButSafe with a carrot follower, then the BT designer doesn't care about the algorithm, they have their abstractions

Reply:
OK, I find it confusing assigning `FollowPath` to a `controller_name`. I think it would help if the field was named `following_type`, or something like that.

Reply:
I disagree with the word type, in pluginlib `type` means something

```
<library path="nav2_navfn_planner">
	<class name="nav2_navfn_planner/NavfnPlanner" type="nav2_navfn_planner::NavfnPlanner" base_class_type="nav2_core::GlobalPlanner">
	  <description></description>
	</class>
</library>
```

The `name` is the actual `name` to load, and is map-able. the `type` is the actual registered plugin class. Using `type` here would mix definitions